contracts/dao/FeiDAO.sol

```diff
@@ -0,0 +1,156 @@
+    uint256 private _proposalThreshold = 2_500_000e18;
+
+    constructor(ERC20VotesComp tribe, ICompoundTimelock timelock)
+        GovernorVotesComp(tribe)
+        GovernorTimelockCompound(timelock)
+        Governor("Fei DAO")
+    {}
+
+    /*
+     * Events to track params changes
+     */
+    event QuorumUpdated(uint256 oldQuorum, uint256 newQuorum);
+    event VotingDelayUpdated(uint256 oldVotingDelay, uint256 newVotingDelay);
+    event VotingPeriodUpdated(uint256 oldVotingPeriod, uint256 newVotingPeriod);
+    event ProposalThresholdUpdated(uint256 oldProposalThreshold, uint256 newProposalThreshold);
+
+    function votingDelay() public view override returns (uint256) {
+        return _votingDelay;
+    }
+
+    function votingPeriod() public view override returns (uint256) {
+        return _votingPeriod;
+    }
+
+    function quorum(uint256)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        return _quorum;
+    }
+
+    function proposalThreshold() public view override returns (uint256) {
+        return _proposalThreshold;
+    }
+
+    // governance setters
+    function setVotingDelay(uint256 newVotingDelay) public onlyGovernance {
+        uint256 oldVotingDelay = _votingDelay;
+        _votingDelay = newVotingDelay;
+        emit VotingDelayUpdated(oldVotingDelay, newVotingDelay);
+    }
+
+    function setVotingPeriod(uint256 newVotingPeriod) public onlyGovernance {
+        uint256 oldVotingPeriod = _votingPeriod;
+        _votingPeriod = newVotingPeriod;
+        emit VotingPeriodUpdated(oldVotingPeriod, newVotingPeriod);
+    }
+
+    function setQuorum(uint256 newQuorum) public onlyGovernance {
+        uint256 oldQuorum = _quorum;
+        _quorum = newQuorum;
+        emit QuorumUpdated(oldQuorum, newQuorum);
+    }
+
+    function setProposalThreshold(uint256 newProposalThreshold) public onlyGovernance {
+        uint256 oldProposalThreshold = _proposalThreshold;
+        _proposalThreshold = newProposalThreshold;
+        emit ProposalThresholdUpdated(oldProposalThreshold, newProposalThreshold);
+    }
+
+    // The following functions are overrides required by Solidity.
+    function getVotes(address account, uint256 blockNumber)
+        public
+        view
+        override(IGovernor, GovernorVotesComp)
+        returns (uint256)
+    {
+        return super.getVotes(account, blockNumber);
+    }
+
+    function state(uint256 proposalId)
+        public
+        view
+        override(IGovernor, Governor, GovernorTimelockCompound)
+        returns (ProposalState)
+    {
+        return super.state(proposalId);
+    }
+
+    function propose(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description
+    )
+        public
+        override(IGovernor, Governor, GovernorCompatibilityBravo)
+        returns (uint256)
+    {
+        return super.propose(targets, values, calldatas, description);
```

Comment:
Does this function need to be permissioned or does the function we call here enforce that?

Reply:
the supercall checks proposal threshold